syntax/scan.go

```diff
@@ -29,8 +30,9 @@ const (
 	OUTDENT
 
 	// Tokens with values
-	IDENT  // x
-	INT    // 123
+	IDENT // x
+	INT   // 123
+	BIGINT
```

Comment:
We don't need a new token type for big integers; re-use the INT token type (since as far as the parser is concerned, they're all integers).  Just allow the data value associated with an INT token to be an int or a *big.Int, and use the nilness of the bigInt field to discriminate these two cases.